controllers/controllers/cluster_controller_test.go

```diff
@@ -0,0 +1,378 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	"github.com/aws/eks-anywhere/pkg/providers/vsphere"
+	"github.com/aws/eks-anywhere/pkg/providers/vsphere/mocks"
+	"github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+var (
+	name      = "test-cluster"
+	namespace = "default"
+)
+
+func TestClusterReconcilerSkipManagement(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	govcClient := mocks.NewMockProviderGovcClient(ctrl)
+
+	secret := createSecret()
+	cluster := createCluster()
+	datacenterConfig := createDataCenter(cluster)
+	bundle := createBundle(cluster)
+	machineConfigCP := createCPMachineConfig()
+	machineConfigWN := createWNMachineConfig()
+
+	objs := []runtime.Object{cluster, datacenterConfig, secret, bundle, machineConfigCP, machineConfigWN}
+
+	s := scheme.Scheme
+	s.AddKnownTypes(anywherev1.GroupVersion, cluster)
+	s.AddKnownTypes(anywherev1.GroupVersion, datacenterConfig)
+	s.AddKnownTypes(anywherev1.GroupVersion, bundle)
+	s.AddKnownTypes(anywherev1.GroupVersion, machineConfigCP)
+
+	cb := fake.NewClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	validator := vsphere.NewValidator(govcClient, &networkutils.DefaultNetClient{})
+	defaulter := vsphere.NewDefaulter(govcClient)
+
+	r := &ClusterReconciler{
+		client:    cl,
+		log:       logf.Log,
+		validator: validator,
+		defaulter: defaulter,
+	}
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	ctx := context.Background()
+	govcClient.EXPECT().ValidateVCenterSetupMachineConfig(ctx, gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(0)
+	govcClient.EXPECT().SearchTemplate(ctx, datacenterConfig.Spec.Datacenter, gomock.Any()).Return("test", nil).Times(0)
+	govcClient.EXPECT().GetTags(ctx, machineConfigCP.Spec.Template).Return([]string{"os:ubuntu", fmt.Sprintf("eksdRelease:%s", bundle.Spec.VersionsBundles[0].EksD.Name)}, nil).Times(0)
+	govcClient.EXPECT().GetWorkloadAvailableSpace(ctx, machineConfigCP.Spec.Datastore).Return(100.0, nil).Times(2).Times(0)
+
+	_, err := r.Reconcile(ctx, req)
+	if err != nil {
+		t.Fatalf("reconcile: (%v)", err)
+	}
+
+	apiCluster := &anywherev1.Cluster{}
+
+	err = r.client.Get(context.TODO(), req.NamespacedName, apiCluster)
+	if err != nil {
+		t.Fatalf("get cluster: (%v)", err)
+	}
+	if apiCluster.Status.FailureMessage != nil {
+		t.Errorf("Expected failure message to be nil")
```

Comment:
nit: include the failure reason in the error message?

Reply:
done